compactor/src/compact.rs

```diff
@@ -460,20 +467,39 @@ impl Compactor {
         Ok(())
     }
 
-    // Group given parquet files into partition of the same (sequencer_id, table_id, partition_id)
-    fn group_parquet_files_into_partition(
-        parquet_files: Vec<ParquetFile>,
-    ) -> BTreeMap<TablePartition, Vec<ParquetFile>> {
-        let mut groups: BTreeMap<TablePartition, Vec<ParquetFile>> = BTreeMap::default();
-        for file in parquet_files {
-            let key = TablePartition::new(file.sequencer_id, file.table_id, file.partition_id);
-            if let Some(val) = groups.get_mut(&key) {
-                val.push(file);
+    // Group time-contiguous non-overlapped groups if their total size is smaller than a threshold
+    fn group_small_contiguous_groups(
+        file_groups: &mut Vec<GroupWithMinTimeAndSize>,
+    ) -> Vec<Vec<ParquetFile>> {
+        let mut groups = Vec::with_capacity(file_groups.len());
+        if file_groups.is_empty() {
+            return groups;
+        }
+
+        // Sort the groups by thier min_tim
+        file_groups.sort_by(|a, b| a.min_time.cmp(&b.min_time));
+
+        let mut current_group = vec![];
+        let mut current_size = 0;
+        for g in file_groups {
+            if current_size + g.total_file_size_bytes < INFLUXDB_IOX_COMPACTION_MAX_SIZE_BYTES {
+                // Group this one with the current_group
+                current_group.extend(g.parquet_files.clone());
```

Comment:
This clone isn't needed; we can move the parquet files into the group:

```suggestion
                current_group.extend(g.parquet_files);
```